app/src/main/java/com/github/se/cyrcle/model/map/MapViewModel.kt

```diff
@@ -45,6 +52,16 @@ class MapViewModel : ViewModel() {
     _cameraPosition.value = cameraState
   }
 
+  /**
+   * Update the focus mode
+   *
+   * @param focusMode the new focus mode This function is used to update the focus mode of the map
+   *   screen. The focus mode is used to center the map on the user's location
+   */
+  fun updateFocusMode(focusMode: Boolean) {
+    _focusMode.value = focusMode
+  }
+
```

Comment:
A `toggleFocusMode()` function would be welcome, as most of the time, we don't want to bother with actually updating, rather just switching the value.

Reply:
I chose this name to stay consisten with the rest of the function of the mapViewModel